From 9fbe98ae6c08a1a981c66af8d6c3cca039ec6abc Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:45 -0400
Subject: [fs] btrfs: fix static checker warnings

Message-id: <1407179109-13714-39-git-send-email-sandeen@redhat.com>
Patchwork-id: 87024
O-Subject: [RHEL7.1 PATCH 038/422] btrfs: fix static checker warnings
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Jeff Mahoney <jeffm@suse.com>

commit 1b8e5df6d9b676f6d31fb098ffdc7d18732729d7
Author: Jeff Mahoney <jeffm@suse.com>
Date:   Wed Nov 20 16:50:23 2013 -0500

    btrfs: fix static checker warnings

    This patch fixes the following warnings:
    fs/btrfs/extent-tree.c:6201:12: sparse: symbol 'get_raid_name' was not declared. Should it be static?
    fs/btrfs/extent-tree.c:8430:9: error: format not a string literal and no format arguments [-Werror=format-security] get_raid_name(index));

    Signed-off-by: Jeff Mahoney <jeffm@suse.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index fe651f4..f08f6dd 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -6195,7 +6195,7 @@ static const char *btrfs_raid_type_names[BTRFS_NR_RAID_TYPES] = {
  [BTRFS_RAID_RAID6] = "raid6",
 };
 
-const char *get_raid_name(enum btrfs_raid_types type)
+static const char *get_raid_name(enum btrfs_raid_types type)
 {
  if (type >= BTRFS_NR_RAID_TYPES)
   return NULL;
@@ -8423,7 +8423,7 @@ static void __link_block_group(struct btrfs_space_info *space_info,
 
   kobject_get(&space_info->kobj); /* put in release */
   ret = kobject_init_and_add(kobj, &btrfs_raid_ktype,
-        &space_info->kobj,
+        &space_info->kobj, "%s",
         get_raid_name(index));
   if (ret) {
    pr_warn("btrfs: failed to add kobject for block cache. ignoring.\n");
-- 
1.7.1